src/Gui/Clipping.cpp

```diff
@@ -86,65 +90,89 @@ class Clipping::Private
         clipView->unref();
         delete sensor;
     }
+
     static void moveCallback(void* data, SoSensor* sensor)
     {
         Q_UNUSED(sensor);
         auto self = static_cast<Private*>(data);
-        if (self->view) {
-            Gui::View3DInventorViewer* view = self->view->getViewer();
+        Gui::View3DInventorViewer const* viewer = self->view.data() ? self->view->getViewer() : nullptr;
+        if (viewer) {
             SoClipPlane* clip = self->clipView;
             SbPlane pln = clip->plane.getValue();
-            clip->plane.setValue(SbPlane(view->getViewDirection(), pln.getDistanceFromOrigin()));
+            clip->plane.setValue(SbPlane(viewer->getViewDirection(), pln.getDistanceFromOrigin()));
+        }
+        else {
+            self->sensor->unschedule();
+            self->isSensing = false;
         }
     }
 };
 
+Ui_Clipping Clipping::Private::ui;
+
+Clipping::Private* Clipping::Private::activeClip = nullptr;
+
 /* TRANSLATOR Gui::Dialog::Clipping */
 
-Clipping::Clipping(Gui::View3DInventor* view, QWidget* parent)
+Clipping::Clipping(Gui::View3DInventor const* view, QWidget* parent)
     : QDialog(parent)
-    , d(new Private)
 {
     // create widgets
-    d->ui.setupUi(this);
+    Private::ui.setupUi(this);
+
+    Private::ui.clipView->setRange(-INT_MAX, INT_MAX);
+    Private::ui.clipView->setSingleStep(0.1f);
+    Private::ui.clipX->setRange(-INT_MAX, INT_MAX);
+    Private::ui.clipX->setSingleStep(0.1f);
+    Private::ui.clipY->setRange(-INT_MAX, INT_MAX);
+    Private::ui.clipY->setSingleStep(0.1f);
+    Private::ui.clipZ->setRange(-INT_MAX, INT_MAX);
+    Private::ui.clipZ->setSingleStep(0.1f);
+
+    Private::ui.dirX->setRange(-INT_MAX, INT_MAX);
+    Private::ui.dirX->setSingleStep(0.1f);
+    Private::ui.dirY->setRange(-INT_MAX, INT_MAX);
+    Private::ui.dirY->setSingleStep(0.1f);
+    Private::ui.dirZ->setRange(-INT_MAX, INT_MAX);
+    Private::ui.dirZ->setSingleStep(0.1f);
+    Private::ui.dirZ->setValue(1.0f);
```

Comment:
The `d` pointer solution is quite common one, why change it to static variable? If anything I'd ditch the `d` pointer and `Private` class as they are not really needed as we don't need to care about public ABI.

Private classes are only needed for libraries - I don't think that we are in this basket.

If we need class to contain the dialog as now they can be multiple instances of the dialog per view - let's add a custom dialog class that will encapsulate the logic and create one instance of such class per 3D View.